exonum/src/runtime/error/error_match.rs

```diff
@@ -128,6 +94,38 @@ impl ErrorMatch {
     }
 }
 
+impl PartialEq<ErrorMatch> for ExecutionError {
+    fn eq(&self, error_match: &ErrorMatch) -> bool {
+        let kind_matches = self.kind == error_match.kind;
+        let runtime_matches = match (error_match.runtime_id, self.runtime_id) {
+            (None, _) => true,
+            (Some(match_id), Some(id)) => match_id == id,
+            _ => false,
+        };
+        let instance_matches = match (error_match.instance_id, &self.call_site) {
+            (None, _) => true,
+            (Some(match_id), Some(CallSite { instance_id, .. })) => match_id == *instance_id,
+            _ => false,
+        };
+        let call_type_matches = match (&error_match.call_type, &self.call_site) {
+            (None, _) => true,
+            (Some(match_type), Some(CallSite { call_type, .. })) => match_type == call_type,
+            _ => false,
+        };
+        kind_matches
+            && runtime_matches
+            && instance_matches
+            && call_type_matches
+            && error_match.description.matches(&self.description)
+    }
+}
+
+impl PartialEq<ExecutionError> for ErrorMatch {
+    fn eq(&self, other: &ExecutionError) -> bool {
+        other.eq(self)
+    }
+}
+
 pub(super) enum StringMatch {
```

Comment:
Nit: It seems possible to move `ErrorMatch` to this module and make `StringMatch` private, if a single test covering matching functionality is moved in this module, too.